util/src/main/scala/geotrellis/util/StreamingByteReader.scala

```diff
@@ -155,4 +189,40 @@ object StreamingByteReader {
 
   def apply(rangeReader: RangeReader, chunkSize: Int): StreamingByteReader =
     new StreamingByteReader(rangeReader, chunkSize)
+
+
+  private[util]
+  def intervalBounds(i: Interval[Long]): (Long, Long) = (
+    i.lowerBound match {
+      case Closed(a) => a
+      case Open(a) => a + 1
```

Comment:
This is more of a genera question than an issue, but when would the `Interval`s have `Open` bounds? In this situation, it seems like they should always be closed.

Reply:
Thats a good question. They can apparently be a result of the difference `\`, taking one interval away from the other other:

```scala
scala> import spire.math._
import spire.math._

scala> import spire.implicits._
import spire.implicits._

scala> Interval(1,10) \ Interval(7,25)
res0: List[spire.math.Interval[Int]] = List([1, 7))
```
Because `7` is on the right, it can't be on the left, so the interval includes things up to `7` but not including `7`. 

This doesn't quite square with the type parameter of `Long`, you kind of know that if its not `7` it must be `6`. I guess this could be useful if you were using integers to model real intervals or more likely encoding the distinction at the type level didn't seem like it was worth while to the library author.